File: emp.py
import os
import time

def run_command(command):
    """Run a command in the terminal."""
    os.system(command)

def start_services():
    """Start Nginx and MySQL."""
    os.system("clear")
    print("Starting The Nginx with MySQL...")
    print("--------------------------------->")
    run_command("sudo systemctl enable nginx mysql.service")
    
    run_command("sudo systemctl start nginx mysql.service")
    print("---------------------------------")
    run_command("sudo systemctl status nginx")
    print("---------------------------------")
    run_command("sudo systemctl status mysql.service")


def stop_services():
    """Stop Nginx and MySQL."""
    os.system("clear")
    print("Stopping The Nginx and MySQL...")
    print("--------------------------------->")
    run_command("sudo systemctl stop nginx mysql.service")
    
    run_command("sudo systemctl disable nginx mysql.service")
    print("---------------------------------")
    run_command("sudo systemctl status nginx")
    print("---------------------------------")
    run_command("sudo systemctl status mysql.service")

def sites_enable_enginx():
    os.system("clear")
    """Monitoring Nginx using multitail """
    print("Checking Sites-Enable Config...")
    print("---------------------------------")
    time.sleep(0.5)
    run_command("ls -all /etc/nginx/sites-enabled")
    time.sleep(0.5)
    print("\nChecking Preps...")
    print("---------------------------------")
    run_command("sudo nginx -t")
    print("---------------------------------")
    print(" ")
    """kill the function and return to the main loop in main()"""
    while True:  # Loop to ensure input is valid
        sub_choice = input("\033[F Press 'q' to go back to menu: ")

        if sub_choice == "q":
            break  # If input is 'q', exit the loop and return to main
        else:
            print("\033[F Invalid, Please Try Again....")
            time.sleep(0.7)

def check_services():
    """Check Nginx and MySQL."""
    os.system("clear")
    print("Checking The Nginx and MySQL...")
    time.sleep(1.7)
    print("---------------------------------")
    run_command("sudo systemctl status nginx")
    time.sleep(0.8)
    print("---------------------------------")
    run_command("sudo systemctl status mysql.service")
    
def monitor_nginx():
    """Monitoring Nginx using multitail """
    os.system("clear")
    print("Monitoring Nginx...")
    time.sleep(1)
    run_command("sudo multitail /var/log/nginx/access.log /var/log/nginx/error.log")


def main():
    """Main menu of the program."""
    while True:
        os.system('clear')
        print("+===========+")
        print("Nginx, Mysql Panel")
        print("-------------")
        print("EMP-Menu")
        print("+===========+")
        print("1. Turn On Engine")
        print("2. Turn Off Engine")
        print("3. Check Preps")
        print("4. Check Engine")
        # If you want to use option number 4,
        # make sure you have previously used option number 1, or 1 then 2
        # or option number 4 will look like it's not working properly
        print("5. View Logs")
        print("0. Bail Out")
        print("-------------")
        print("Always press 'q' to back menu")
        print("+===========+")
        
        choice = input("Choose an option (1/2/3/4/5/0): ")
        
        if choice == '1':
            start_services()
        elif choice == '2':
            stop_services()
        elif choice == '3':
            sites_enable_enginx()
        elif choice == '4':
            check_services()
        elif choice == '5':
            monitor_nginx()
        elif choice == '0':
            print("Exiting the program.")
            time.sleep(0.4)
            os.system('clear')
            break
        else:
            print("Invalid choice. Please try again.")
            time.sleep(0.4)

if __name__ == "__main__":
    main()

File: emp.go
package main

import (
	"fmt"
	"os"
	"os/exec"
	"time"
)

func runCommand(command string) {
	// Execute commands in terminal
	cmd := exec.Command("bash", "-c", command)
	cmd.Stdout = os.Stdout
	cmd.Stderr = os.Stderr
	if err := cmd.Run(); err != nil {
		fmt.Println("Error:", err)
	}
}

func clearScreen() {
	fmt.Print("\033[H\033[2J")
}

func startServices() {
	clearScreen()
	fmt.Println("Starting The Nginx with MySQL...")
	fmt.Println("--------------------------------->")
	runCommand("sudo systemctl enable nginx mysql.service")
	runCommand("sudo systemctl start nginx mysql.service")
	fmt.Println("---------------------------------")
	runCommand("sudo systemctl status nginx")
	fmt.Println("---------------------------------")
	runCommand("sudo systemctl status mysql.service")
}

func stopServices() {
	clearScreen()
	fmt.Println("Stopping The Nginx and MySQL...")
	fmt.Println("--------------------------------->")
	runCommand("sudo systemctl stop nginx mysql.service")
	runCommand("sudo systemctl disable nginx mysql.service")
	fmt.Println("---------------------------------")
	runCommand("sudo systemctl status nginx")
	fmt.Println("---------------------------------")
	runCommand("sudo systemctl status mysql.service")
}

func sitesEnableEnginx() {
		clearScreen() // Clear screen every time you enter the menu
		fmt.Println("Checking Sites-Enable Config...")
		fmt.Println("---------------------------------")
		time.Sleep(500 * time.Millisecond)
		runCommand("ls -al /etc/nginx/sites-enabled")
		time.Sleep(500 * time.Millisecond)
		fmt.Println("\nChecking Preps...")
		fmt.Println("---------------------------------")
		runCommand("sudo nginx -t")
		fmt.Println("---------------------------------")
		fmt.Println(" ")
		
		var subChoice string

		for {
			fmt.Print("\033[F Press 'q' to go back to menu: ")
			fmt.Scan(&subChoice)

			if subChoice == "q" {
				break
			} else {
				fmt.Println("\033[F Invalid, Please Try Again....")
				time.Sleep(700 * time.Millisecond)
				}
			}
}

func checkServices() {
	clearScreen()
	fmt.Println("Checking The Nginx and MySQL...")
	time.Sleep(1700 * time.Millisecond)
	fmt.Println("---------------------------------")
	runCommand("sudo systemctl status nginx.service")
	time.Sleep(800 * time.Millisecond)
	fmt.Println("---------------------------------")
	runCommand("sudo systemctl status mysql.service")
}

func monitorNginx() {
	clearScreen()
	fmt.Println("Monitoring Nginx...")
	time.Sleep(1000 * time.Millisecond)
	runCommand("sudo multitail /var/log/nginx/access.log /var/log/nginx/error.log")
}

func main() {
	for {
		clearScreen()
		fmt.Println("+===========+")
		fmt.Println("Nginx, Mysql Panel")
		fmt.Println("-------------")
		fmt.Println("EMP-Menu")
		fmt.Println("+===========+")
		fmt.Println("1. Turn On Engine")
		fmt.Println("2. Turn Off Engine")
		fmt.Println("3. Check Preps")
		fmt.Println("4. Check Engine")
		// If you want to use option number 4,
		// make sure you have previously used option number 1, or 1 then 2
		// or option number 4 will look like it's not working properly
		fmt.Println("5. View Logs")
		fmt.Println("0. Bail Out")
		fmt.Println("-------------")
		fmt.Println("Always press 'q' to back menu")
		fmt.Println("+===========+")

		var choice string
		fmt.Print("Choose an option (1/2/3/4/5/0): ")
		fmt.Scan(&choice)

		switch choice {
		case "1":
			startServices()
		case "2":
			stopServices()
		case "3":
			sitesEnableEnginx()
		case "4":
			checkServices()
		case "5":
			monitorNginx()
		case "0":
			fmt.Println("Exiting the program.")
			time.Sleep(400 * time.Millisecond)
			clearScreen()
			return
		default:
			fmt.Println("Invalid choice. Please try again.")
			time.Sleep(400 * time.Millisecond)
		}
	}
}

